### DISCUSSION TITLE
[QUESTION] How to debug issues with automatic deletion of workloads on unexpectedly detached volumes

### DISCUSSION THREAD

**Discussion by fh-fritz** [longhorn/longhorn#6538] [General]
## Question
I'm experiencing issues in my k8s cluster with postgres (zalando operator) and also prometheus were suddenly the PVCs switch to read only. If i delete the pod by hand it works. PVCs are ReadWriteOnce

when i look in longhorn ui the following flag is enabled:
https://longhorn.io/docs/archives/1.1.0/references/settings/#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly

**which data I can gather to create a proper issue?** I can't reproduce it easily it happens after some time (once every (2) week(s))

## Environment

 - Longhorn version: longhornio/longhorn-engine:v1.4.2
 - Kubernetes version: v1.27.2 and v1.27.3
 - Node config
   - OS type and version: Ubuntu 22.04.2 LTS
   - Kernel version: 5.15.0-75-generic
   - CPU per node: 4
   - Memory per node: 4GB
   - Disk type: ssd
   - Network bandwidth and latency between the nodes: ~2ms and ~20ms (depending on nodes, they are split between hosting providers)
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): kvm/vps

## Additional context

example `mount` from prometheus 
```
/prometheus $ mount | grep ro
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
sysfs on /sys type sysfs (ro,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup type cgroup2 (ro,nosuid,nodev,noexec,relatime)
/dev/mapper/vg0-root on /etc/prometheus type ext4 (ro,relatime)
/dev/mapper/vg0-root on /etc/hosts type ext4 (rw,relatime)
/dev/mapper/vg0-root on /dev/termination-log type ext4 (rw,relatime)
/dev/mapper/vg0-root on /etc/prometheus/rules type ext4 (ro,relatime)
/dev/longhorn/pvc-c2ca73d9-0631-4875-9710-b843cbff8ab6 on /var/lib/prometheus type ext4 (ro,relatime)
tmpfs on /var/run/secrets/kubernetes.io/serviceaccount type tmpfs (ro,relatime,size=3902948k,inode64)
proc on /proc/bus type proc (ro,nosuid,nodev,noexec,relatime)
proc on /proc/fs type proc (ro,nosuid,nodev,noexec,relatime)
proc on /proc/irq type proc (ro,nosuid,nodev,noexec,relatime)
proc on /proc/sys type proc (ro,nosuid,nodev,noexec,relatime)
proc on /proc/sysrq-trigger type proc (ro,nosuid,nodev,noexec,relatime)
tmpfs on /proc/acpi type tmpfs (ro,relatime,inode64)
tmpfs on /proc/kcore type tmpfs (rw,nosuid,size=65536k,mode=755,inode64)
tmpfs on /proc/keys type tmpfs (rw,nosuid,size=65536k,mode=755,inode64)
tmpfs on /proc/timer_list type tmpfs (rw,nosuid,size=65536k,mode=755,inode64)
tmpfs on /proc/scsi type tmpfs (ro,relatime,inode64)
tmpfs on /sys/firmware type tmpfs (ro,relatime,inode64)
/prometheus $
```

**Comment by derekbit** [comment]
> Network bandwidth and latency between the nodes: ~2ms and ~20ms (depending on nodes, they are split between hosting providers)

1 Gb/s or 10 Gb/s NIC?

Can you provide a support bundle?

**Comment by fh-fritz** [comment]
> > Network bandwidth and latency between the nodes: ~2ms and ~20ms (depending on nodes, they are split between hosting providers)
> 
> 1 Gb/s or 10 Gb/s NIC?

I guess its lower than 1Gbit between some nodes because the are in different data centers.
However there are just 3 nodes and so all PVCs are replicated to all nodes where pods are running.

I think this is irrelevant here because of what reason deletion of pods should not work when the connection is slow?
It's realistic that volumes detach because of latency, but this should cause the pods using them to be recreated (like described in the link). I did not find this setting in a config map however, just in the longhorn UI
> 
> Can you provide a support bundle?

I don't know what you mean but I'm sure there isn't any.
I don't want support for the issues of disconnecting PVCs, instead I want to gather all relevant configs, logs and data to provide it for a proper bug report that someone can investigate why the longhorn feature is not working like expected. The issue is that I don't know which logs/configs/... are relevant, therefore this question/issue

>Please refer to the support bundle section in the official doc.
>https://longhorn.io/docs/1.4.2/advanced-resources/support-bundle/

**Comment by fh-fritz** [comment]
> Please refer to the support bundle section in the official doc.
> https://longhorn.io/docs/1.4.2/advanced-resources/support-bundle/

I created a support bundle, however I manually resolved the issue currently. Is it still helpful then?

Where should I upload/send the file (its 90mb big)? since I likely contains sensitive data


> 1 Gb/s or 10 Gb/s NIC?

regarding this question, I read the bandwith limits here:
https://longhorn.io/kb/troubleshooting-volume-readonly-or-io-error/#root-causes

 the db is quite idle since its a staging environment

**Comment by derekbit** [comment]
> > Please refer to the support bundle section in the official doc.
> > https://longhorn.io/docs/1.4.2/advanced-resources/support-bundle/
> 
> I created a support bundle, however I manually resolved the issue currently. Is it still helpful then?
> 
> Where should I upload/send the file (its 90mb big)? since I likely contains sensitive data

You can send it to longhorn-support-bundle@suse.com.

> 
> > 1 Gb/s or 10 Gb/s NIC?
> 
> regarding this question, I read the bandwith limits here: https://longhorn.io/kb/troubleshooting-volume-readonly-or-io-error/#root-causes
> 
> the db is quite idle since its a staging environment

What is the `db` you mentioned?

**Comment by fh-fritz** [comment]
> You can send it to [longhorn-support-bundle@suse.com](mailto:longhorn-support-bundle@suse.com).

I created a new bundle now while the issue happend again. I sent it to this mail address as a link

> What is the `db` you mentioned?

like mentioned in the first post a (low load) postgres db run by zalando operator
